src/ecalc/libraries/libecalc/common/libecalc/input/yaml_types/components/legacy/yaml_fuel_consumer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Any, Dict, Type
+
+from libecalc.dto.base import ConsumerUserDefinedCategoryType
+from libecalc.input.yaml_types import YamlBase
+from libecalc.input.yaml_types.components.category import CategoryField
+from libecalc.input.yaml_types.placeholder_type import PlaceholderType
+from libecalc.input.yaml_types.temporal_model import TemporalModel
+from pydantic import Field
+
+
+class YamlFuelConsumer(YamlBase):
```

Comment:
include Legacy or similar in the name of the class as well? Do we have a "non-legacy fuel consumer"? What is the reason why the legacy yaml type is included here? Plan to remove/change it?

Reply:
By using legacy I'm trying to make it clear which files are connected across yaml,dto,core etc. If we end up creating several versions for each type (system v2,v3,v4 etc), then we probably need to group those classes in a similar way (by version, not `legacy`)